product_catalog_redesign/static/src/components/kanban_record.js

```diff
@@ -0,0 +1,31 @@
+/** @odoo-module */
+import { ProductCatalogKanbanRecord } from "@product/product_catalog/kanban_record";
+import { useFileViewer } from "@web/core/file_viewer/file_viewer_hook";
+
+export class ProductCatalogKanbanRecordInherited extends ProductCatalogKanbanRecord {
+    setup() {
+        super.setup();
+        this.fileViewer = useFileViewer();
+    }
+    onGlobalClick(ev) {
+        //Opens pop up dialog box when clicked on product image.
+        if (ev.target.closest(".o_product_image_custom")) {
+            const imageUrl = ev.target.getAttribute("src");
+            this.fileViewer.open(
+                { isImage: true,
+                  isViewable: true,
+                  defaultSource: imageUrl,
+                  downloadUrl: imageUrl 
+                })
+            return;
+        }
+        if (ev.target.closest(".o_product_catalog_cancel_global_click")) {
+            return;
+        }
+        if (this.productCatalogData.quantity === 0) {
+            this.addProduct();
+        } else {
+            this.increaseQuantity();
+        }
```

Comment:
```suggestion
        super.onGlobalClick(ev);
```
What was the reasoning behind doing this?

Reply:
I thought calling super here is not work as i was overriding method here. Thank you the suggestion.